pulp_rpm/app/models/advisory.py

```diff
@@ -100,14 +101,19 @@ class UpdateRecord(Content):
     # A field that represents the hash digest of the update record. Used to track differences
     # between two UpdateRecord objects without having to examine the associations like
     # UpdateCollection or UpdateCollectionPackage.
-    digest = models.TextField(unique=True)
+    digest = models.TextField(db_index=True)
+
+    _pulp_domain = models.ForeignKey("core.Domain", default=get_domain_pk, on_delete=models.PROTECT)
 
     @classmethod
     def natural_key_fields(cls):
         """
         Digest is used as a natural key for UpdateRecords.
         """
-        return ("digest",)
+        return (
+            "_pulp_domain",
+            "digest",
+        )
```

Comment:
Have you considered just removing this function? The standard implementation to use the first "unique_together" tuple should do the trick.

Reply:
Right, I'm not sure why this function is here, maybe there was a historical reason where we needed to use something different than the uniqueness constraint?  But it doesn't seem to be the case presently.

Reply:
All the natural_key_fields() impls were in place prior to this PR. You're def correct that they're overkill - or, worse, they're duplicate code that leads to problems. Removing from advisory/comps/modulemd.